onnxruntime/python/tools/transformers/onnx_model_t5.py

```diff
@@ -29,25 +29,460 @@ def __init__(
+                    return
+                if "past_key_cross" not in past_key:
+                    return
+        else:
+            k_nodes = self.model.match_parent_path(
+                matmul_qk,
+                ["Transpose", "Concat", "Reshape", "MatMul"],
+                [1, 0, 1, 0],
+            )
+            if k_nodes is not None:
+                _, concat_k, reshape_k, _ = k_nodes
+                key = reshape_k.input[0]
+                past_key_transpose_node = output_name_to_node[concat_k.input[0]]
+                past_key = past_key_transpose_node.input[0]
+                if past_key in output_name_to_node:
+                    return
+                if "past_key_self" not in past_key:
+                    return
+                present_key_transpose_nodes = input_name_to_nodes[concat_k.output[0]]
+                for present_key_transpose_node in present_key_transpose_nodes:
+                    # print("present_key_transpose_node:", present_key_transpose_node)
+                    present_key_candidate = self.model.find_graph_output(present_key_transpose_node.output[0])
+                    # print("present_key_candidate:", present_key_candidate)
+                    if present_key_candidate is not None:
+                        present_key = present_key_candidate.name
+                        break
+                if present_key is None:
+                    return
+                if "present_key_self" not in present_key:
+                    return
+            else:
+                k_nodes = self.model.match_parent_path(
+                    matmul_qk,
+                    ["Transpose", "Reshape", "MatMul"],
+                    [1, 0, 0],
+                )
+                if k_nodes is None:
+                    return
+                _, reshape_k, _ = k_nodes
+                key = reshape_k.input[0]
+                present_key_transpose_nodes = input_name_to_nodes[reshape_k.output[0]]
+                for present_key_transpose_node in present_key_transpose_nodes:
+                    present_key_candidate = self.model.find_graph_output(present_key_transpose_node.output[0])
+                    if present_key_candidate is not None:
+                        present_key = present_key_candidate.name
+                        break
+                if present_key is None:
+                    return
+                if "present_key_self" not in present_key:
+                    return
+
+        q_nodes = self.model.match_parent_path(
+            matmul_qk,
+            ["Transpose", "Reshape", "MatMul"],
+            [0, 0, 0],
+        )
+        if q_nodes is None:
+            return
+
+        transpose_q, reshape_q, matmul_q = q_nodes
+
+        if matmul_q.input[0] != input_shape_node.input[0]:
+            return
+
+        q_num_heads, q_hidden_size = self.get_num_heads_and_hidden_size(reshape_q)
+
+        if self.static_kv == 1 and past_key is not None:
```

Comment:
else return?

Reply:
in cross attention the past_* is first assigned to var past_*. since the schema changed the past_* to 'key' and 'value' so we re-assign the input here